pkg/controller/test-images-distributor/test_images_distributor.go

```diff
@@ -72,18 +72,26 @@ func AddToManager(mgr manager.Manager,
 	for buildClusterName, buildClusterManager := range buildClusterManagers {
 		buildClusters.Insert(buildClusterName)
 		r.buildClusterClients[buildClusterName] = imagestreamtagwrapper.MustNew(buildClusterManager.GetClient(), buildClusterManager.GetCache())
-		// TODO: Watch buildCluster ImageStreams as well. For now we assume no one will tamper with them.
-		if buildClusterName == "app.ci" {
-			if err := c.Watch(
-				source.NewKindWithCache(&testimagestreamtagimportv1.TestImageStreamTagImport{}, buildClusterManager.GetCache()),
-				testImageStreamTagImportHandler(),
-			); err != nil {
-				return fmt.Errorf("failed to create watch for testimagestreamtagimports: %w", err)
-			}
-		}
 	}
 
-	objectFilter, err := testInputImageStreamTagFilterFactory(log, configAgent, r.buildClusterClients["app.ci"], resolver, additionalImageStreamTags, additionalImageStreams, additionalImageStreamNamespaces)
+	// TODO: Watch buildCluster ImageStreams as well. For now we assume no one will tamper with them.
+	if err := c.Watch(
```

Comment:
IMHO both the `c.Watch` and the `testInputImageStreamTagFilterFactory` should simply be using `registryManager.GetCache()`/`registryManager.GetClient()`